packages-exp/auth-exp/src/core/util/location.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export function getCurrentUrl(): string {
+  return window?.location?.href || self?.location?.href || '';
```

Comment:
Isn't self just a direct alias to window?

Reply:
apparently window doesn't exist in workers but self does, this should probably just use the self version though

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/Window/self